src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +96,15 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                Clipboard.writeTypes(selection || SelectionScraper.getCustomAsTypes(messageHtml));
```

Comment:
Here revert the logic to use the selection and now `Clipboard.setHTML` will be used as explained before. 

IMO, explicitly convert the HTML to MD if needed and set it.

Reply:
We need MD on native, mobile, and firefox, which is how I achieve consistent behavior in the videos below. What do you mean by setting it explicitly? That is resolved in the clipboard based on the platform. Do you suggest testing the platform here in the context menu and everywhere where it is needed? That would introduce a lot of duplicate code.

Reply:
Setting MD is equivalent to setting text as it will use `Clipboard.setString`. I don't think it will as there is only only one clipboard Action that is using it.  

You can create a function to group the code and use it. But falling back to MD where the function says writeTypes, does not seem right. Code only supports writing MD on native and still, the function says `writeTypes`. 

Do you have ideas to refactor the code?

Reply:
I see. OK, I'll try to improve it.